### ISSUE TITLE
Compiling Error in SSR (Next JS)

### ISSUE THREAD

**Issue by NimaKaviyani** [mberneti/react-datepicker2#69]
Hi,

i see this bug in next js like this issue #60 :
```
> Build error occurred
C:\nginx\html\aaaa-nextjs\node_modules\react-datepicker2\dist\index.js:1
import momentJalaali from 'moment-jalaali';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Module._compile (internal/modules/cjs/loader.js:895:18)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.KOFR (C:\nginx\html\aaaa-nextjs\.next\server\static\KblQtgqMCx70DTwxzFT68\pages\admin\events\edit\[id].js:904:18)
    at __webpack_require__ (C:\nginx\html\aaaa-nextjs\.next\server\static\KblQtgqMCx70DTwxzFT68\pages\admin\events\edit\[id].js:23:31)
    at Module.Tzl5 (C:\nginx\html\aaaa-nextjs\.next\server\static\KblQtgqMCx70DTwxzFT68\pages\admin\events\edit\[id].js:1012:35)
    at __webpack_require__ (C:\nginx\html\aaaa-nextjs\.next\server\static\KblQtgqMCx70DTwxzFT68\pages\admin\events\edit\[id].js:23:31) {
  type: 'SyntaxError'
}
```
next version: 9.3.0
react-datepicker2 version: 3.0.15

**Comment by afsanefda** [comment]
I have the same problem using the latest version 3.0.15!

but it is fine in my previous version which is 3.0.3! 
I am using nextjs and conditional imorting is not solving the problem!

**Comment by NimaKaviyani** [comment]
@afsanefda thank you. The version you mentioned is working.

**Comment by mberneti** [comment]
@all-contributors please add @afsanefda for maintenance and question

**Comment by AliNazariii** [comment]
I have same problem.

```
import momentJalaali from 'moment-jalaali';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:1117:16)
    at Module._compile (internal/modules/cjs/loader.js:1165:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Module.require (internal/modules/cjs/loader.js:1090:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at eval (webpack-internal:///react-datepicker2:1:18)
    at Object.react-datepicker2 (/media/ali/Mine/1v1-front/.next/server/static/development/pages/dashboard/plan.js:415:1)
    at __webpack_require__ (/media/ali/Mine/1v1-front/.next/server/static/development/pages/dashboard/plan.js:23:31)
    at eval (webpack-internal:///./components/Dashboard/Planning/Calendar.js:5:75)
    at Module../components/Dashboard/Planning/Calendar.js (/media/ali/Mine/1v1-front/.next/server/static/development/pages/dashboard/plan.js:149:1)
    at __webpack_require__ (/media/ali/Mine/1v1-front/.next/server/static/development/pages/dashboard/plan.js:23:31)
    at eval (webpack-internal:///./components/Dashboard/Planning/Planning.js:5:67)
    at Module../components/Dashboard/Planning/Planning.js (/media/ali/Mine/1v1-front/.next/server/static/development/pages/dashboard/plan.js:161:1)
    at __webpack_require__ (/media/ali/Mine/1v1-front/.next/server/static/development/pages/dashboard/plan.js:23:31)
```

What should I do?

**Comment by TrueMoein** [comment]
Same issue...

**Comment by komailf67** [comment]
import dynamic from "next/dynamic";
const DatePicker = dynamic(() => import("react-datepicker2"), { ssr: false });
const momentJalaali = dynamic(() => import("moment-jalaali"), { ssr: false });

.
.
.
class Test extends Component {

  state = {
    value: "",
  };

**Comment by mberneti** [comment]
@NimaKaviyani @afsanefda @AliNazariii @TrueMoein @komailf67 Thank you for the contribution.
This issue fixed on v3.3.11
